mrmustard/lab_dev/states/base.py

```diff
@@ -266,17 +266,16 @@ def _L2_norms(self) -> RealVector:
         The `L2` norm (squared) of a ``Ket``, or the Hilbert-Schmidt norm of a ``DM``, element-wise along the batch dimension.
         """
         settings.UNSAFE_ZIP_BATCH = True
-        rep = (self >> self.dual).representation
+        rep = self >> self.dual
         settings.UNSAFE_ZIP_BATCH = False
-        return math.real(rep.c if isinstance(rep, Bargmann) else rep.array)
+        return math.real(rep)
 
     @property
     def L2_norm(self) -> float:
         r"""
         The `L2` norm (squared) of a ``Ket``, or the Hilbert-Schmidt norm of a ``DM``.
         """
-        rep = (self >> self.dual).representation
-        return math.sum(math.real(rep.scalar))
+        return math.sum(math.real(self >> self.dual))
```

Comment:
Do we want to specify a MM type for what this returns?

Reply:
the return type is actually `float`